sway/main.c

```diff
@@ -135,6 +135,25 @@ void detect_proprietary() {
 				"You need nvidia, nvidia_modeset, nvidia_uvm, and nvidia_drm."
 				"\x1B[0m\n");
 		}
+#ifdef __linux__
+        f = fopen("/sys/module/nvidia_drm/parameters/modeset", "r");
+        if (f) {
+            char *line = read_line(f);
+            if (line && strstr(line, "Y")) {
+                // nvidia-drm.modeset is set to 0
+				fprintf(stderr, "\x1B[1;31mWarning: You must load "
+					"nvidia-drm with the modeset option on to use "
+					"the proprietary driver.\x1B[0m\n");
```

Comment:
This feedback should be more actionable.

Reply:
Updated. I used something similar to the original help string since it shouldn't be affected by the users choice of init system